airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/up_to_date/pipeline.py

```diff
@@ -0,0 +1,326 @@
+from pipelines.airbyte_ci.connectors.bump_version.pipeline import AddChangelogEntry, BumpDockerImageTagInMetadata, get_bumped_version
+from pipelines.airbyte_ci.connectors.consts import CONNECTOR_TEST_STEP_ID
+from pipelines.airbyte_ci.connectors.context import ConnectorContext, PipelineContext
+from pipelines.airbyte_ci.connectors.reports import ConnectorReport, Report
+from pipelines.consts import LOCAL_BUILD_PLATFORM
+from pipelines.dagger.actions.python.common import with_python_connector_installed
+from pipelines.helpers.execution.run_steps import STEP_TREE, StepToRun, run_steps
+from pipelines.helpers.utils import sh_dash_c
+from pipelines.models.steps import Step, StepResult, StepStatus
+
+if TYPE_CHECKING:
+    from typing import Iterable, List, Optional
+
+    from anyio import Semaphore
+
+PACKAGE_NAME_PATTERN = r"^([a-zA-Z0-9_.\-]+)(?:\[(.*?)\])?([=~><!]=?[a-zA-Z0-9\.]+)?$"
+
+
+class CheckIsUpdateCdkCandidate(Step):
+    """Check if the connector is a candidate for migration to poetry.
+    Candidate conditions:
+    - The connector is a Python connector.
+    - The connector is a source connector.
+    - The connector is using poetry.
+    - The connector has a base image defined in the metadata.
+    """
+
+    context: ConnectorContext
+
+    title = "Check if the connector is a candidate for CDK upgrade."
+
+    def __init__(self, context: PipelineContext, cdk_version: str) -> None:
+        super().__init__(context)
+        self.cdk_version = cdk_version
+
+    async def _run(self) -> StepResult:
+        connector_dir_entries = await (await self.context.get_connector_dir()).entries()
+        if self.context.connector.language not in [ConnectorLanguage.PYTHON, ConnectorLanguage.LOW_CODE]:
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector is not a Python connector.",
+            )
+        if self.context.connector.connector_type != "source":
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector is not a source connector.",
+            )
+        if not "poetry.lock" in connector_dir_entries or not "pyproject.toml" in connector_dir_entries:
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector requires poetry.",
+            )
+
+        if not self.context.connector.metadata or not self.context.connector.metadata.get("connectorBuildOptions", {}).get("baseImage"):
+            return StepResult(
+                step=self,
+                status=StepStatus.SKIPPED,
+                stderr="The connector CDK can't be updated because it does not have a base image defined in the metadata.",
+            )
+
+        # TODO: is there a fast way to check if the connector is already using the latest CDK version?
```

Comment:
Poetry has some tools to play around with for this. `poetry show -o | awk '{print $1}' | grep airbyte-cdk` will work. I didn't immediately see a way to make poetry emit json but might have missed it.

Reply:
I did that below. I didn't know if you could do that outside of the container.